sources/packages/api/src/route-controllers/restriction/restriction.aest.controller.ts

```diff
@@ -164,14 +166,16 @@ export class RestrictionController extends BaseController {
    * @param studentId
    * @param payload
    */
-  @Groups(UserGroups.AESTUser)
-  @AllowAuthorizedParty(AuthorizedParties.aest)
   @Roles(Role.StudentAddRestriction)
+  @ApiUnprocessableEntityResponse({
+    description:
+      "The given restriction type is not Provincial. Only provincial restrictions can be added.",
+  })
   @Post("/student/:studentId")
   async addStudentProvincialRestriction(
     @UserToken() userToken: IUserToken,
     @Param("studentId") studentId: number,
-    @Body() payload: AssignRestrictionDTO,
+    @Body() payload: AssignRestrictionAPIInDTO,
   ): Promise<void> {
     const restriction =
       await this.restrictionService.getProvincialRestrictionById(
```

Comment:
The method `getProvincialRestrictionById` is not providing the enum values as parameters. Can we change it in this PR?
![image](https://user-images.githubusercontent.com/61259237/184966685-50dd33c6-4fdd-40c1-847b-ad6263e55e89.png)